app/workers/copay_notifications/parse_new_statements_job.rb

```diff
@@ -6,14 +6,22 @@ class ParseNewStatementsJob
 
     sidekiq_options retry: false
 
+    # time (in seconds) between scheduling batch of jobs
+    JOB_INTERVAL = Settings.mcp.notifications.job_interval
+    # number of jobs to perform at next interval
+    BATCH_SIZE = Settings.mcp.notifications.batch_size
+
     def perform(statements_json_byte)
       StatsD.increment('api.copay_notifications.json_file.total')
       # Decode and parse large json file (~60-90k objects)
       statements_json = Oj.load(Base64.decode64(statements_json_byte))
       unique_statements = statements_json.uniq { |statement| statement['veteranIdentifier'] }
 
-      unique_statements.each do |statement|
-        CopayNotifications::NewStatementNotificationJob.perform_async(statement)
+      unique_statements.each_with_index do |statement, index|
+        # For every BATCH_SIZE jobs, enqueue the next BATCH_SIZE amount of jobs JOB_INTERVAL seconds later
+        CopayNotifications::NewStatementNotificationJob.perform_in(
+          JOB_INTERVAL * (index / BATCH_SIZE), statement
+        )
```

Comment:
@ScottyRJames I just wanted to make sure this is what you intended. If JOB_INTERVAL and BATCH_SIZE were 90 and 100 respectively, then we'd be kicking off a new job every .9 seconds (not JOB_INTERVAL seconds), correct? Maybe I'm missing something?

Reply:
@ryan-mcneil It will kick off 100 new jobs every 90 seconds rather since ruby division defaults to the floor with integers. 

For example, if one of the statements had index 127 it would be evaluated like `90 * (127 / 100)` -> `90 * 1` which `perform_in` will take and schedule the job at 90 seconds later. So if `unique_statements` were to be 500 statements long each of the slated times would be like

| Indices  | Start Time |
| ------------- | ------------- |
| 0 - 99  | Time.now  |
| 100 - 199  | Time.now + 90s  |
| 200 - 299  | Time.now + 180s |
| 300 - 399  | Time.now + 270s |
| 400 - 499  | Time.now + 360s |
| 500 | Time.now + 450s |

If we wanted to go towards your thinking, it would look like  like `JOB_INTERVAL * (index / BATCH_SIZE).to_f`. I think by default Sidekiq checks for new jobs every 5 seconds so it wouldn't be to the exact millisecond that it's scheduled at. It's still workable if you think it would be better to stagger the jobs even more rather than spinning up a batch at a time but it would be more like 6-7 new jobs every 5 seconds vs a new job every .9 seconds.

Reply:
Yeah, you're right. I wasn't considering the ruby division. Just wanted to make sure you considered it, and it looks like you're a step ahead of me 👌